File: client/src/app/_models/state.ts
import { City } from "./city";
import { Country } from "./country";

export interface State {
  id: number;
  name: string;
  country_Id: number;
  country_Code: string;
  fips_Code: string;
  iso2: string;
  type: string;
  latitude: number | null;
  longitude: number | null;
  created_At: Date | null;
  updated_At: Date;
  flag: boolean;
  wikiDataId: string;
  cities: City[];
  country: Country;
}

File: API/Entities/State.cs
using System.ComponentModel.DataAnnotations;

namespace API.Entities
{
    public class State
    {
        [Key]        
        public int Id { get; set; }
        public string Name { get; set; }
        public int Country_Id { get; set; }
        public string Country_Code { get; set; }
        public string Fips_Code { get; set; }
        public string Iso2 { get; set; }
        public string Type { get; set; }
        public decimal? Latitude { get; set; }
        public decimal? Longitude { get; set; }
        public DateTime? Created_At { get; set; }
        public DateTime Updated_At { get; set; }
        public bool Flag { get; set; }
        public string WikiDataId { get; set; }
        public ICollection<City> Cities {get;set;}
        public Country Country {get;set;}
    }
}